src/oph/heratepalvelu/tep/kestojenUudelleenlaskentaHandler.clj

```diff
@@ -0,0 +1,68 @@
+(ns oph.heratepalvelu.tep.kestojenUudelleenlaskentaHandler
+  "TODO"
+  (:require [clojure.tools.logging :as log]
+            [environ.core :refer [env]]
+            [oph.heratepalvelu.db.dynamodb :as ddb]
+            [oph.heratepalvelu.tep.tepCommon :as tc]
+            [oph.heratepalvelu.tep.niputusHandler :as nh]
+            [oph.heratepalvelu.log.caller-log :refer
+             [log-caller-details-scheduled]]))
+
+(gen-class :name "oph.heratepalvelu.tep.kestojenUudelleenlaskentaHandler"
+           :methods
+           [[^:static handleKestojenUudelleenlaskenta
+             [com.amazonaws.services.lambda.runtime.events.ScheduledEvent
+              com.amazonaws.services.lambda.runtime.Context] void]])
+
+(def rahoituskausi {:alkupvm "2022-07-01" :loppupvm "2023-06-30"})
+
+(defn scan-for-unhandled-nippus!
+  [exclusive-start-key]
+  (let [resp (ddb/scan
+               {:filter-expression
+                (str
+                  "niputuspvm >= :rahoituskausi_alkupvm AND "
+                  "niputuspvm <= :rahoituskausi_loppupvm AND "
+                  "attribute_not_exists(keston_uudelleenlaskenta_suoritettu)")
```

Comment:
Onko meillä mitään syytä muuttaa nippuja vai pitäisikö vaan suoraan käpistellä jaksoja?  Vai onko tää sitä varten, että handler olis paremmin uudelleenkäynnistettävä, että se pystyisi käsittelemään nippuja inkrementaalisesti?

Reply:
Oma alkuperäinen ajatus oli, että jos nippuja on vähemmän kuin jaksoja, tuo `scan` olisi nopeampi operaatio. Kuitenkin, näin jälkikäteen viisaampana, tuo nippujen skannaaminen ei taida olla kovinkaan paljon nopeampi, jos niput sisältävät tyypillisesti vain sen 1-2 jaksoa per nippu. :thinking:

Reply:
Hmm.  No ainakaan siitä ei oo mitään haittaa, että mennään nippu kerrallaan.  Ja jos joka tapauksessa ei oo mitään indeksiä, mistä saisi käsittelemättömät niput/jaksot niin scannilla mennään :/  Yksi mahdollisuus on siis tietysti pitää metadataa siitä, mihin niputuspvm:iin asti on käsitelty, mutta ehkä se ei oo nyt tekemisen arvoista, ellei tuo scan vie minuuttitolkulla ennen kuin palauttaa yhtäkään tulosta.

Reply:
En itekään tykkää tästä, että joudutaan tekemään scan sopivien indeksien puuttuessa. Tuon koko taulun `scan` pyörähtää läpi ~10 sekunnnissa.

Tän lambdan suorituskyky on ehkä verrattavissa tuohon `tpkNiputusHandler`-lambdaan. Tääkin ad-hoc lambda vaatii useampia ajokertoja, mutta uskoakseni kaikki niput pitäisi tulla silti käsitellyksi ihan inhimillisessä ajassa.

Reply:
Päivitetyssä koodissa käpistellään nyt suoraan jaksoja :)